packages/navie/src/services/openai-completion-service.ts

```diff
@@ -279,43 +289,60 @@ export default class OpenAICompletionService implements CompletionService {
     return undefined;
   }
 
-  async *complete(messages: readonly Message[], options?: { temperature?: number }): Completion {
-    const promptTokensPromise = this.countTokens(messages);
+  async *complete(messages: readonly Message[], options?: JsonOptions): Completion {
+    const model = options?.model ?? this.model.modelName;
+    const isO1 = this.modelName.startsWith('o1-');
 
-    const sentMessages = mergeSystemMessages(messages);
+    const sentMessages = mergeSystemMessages(messages, isO1 ? 'user' : 'system');
     for (const message of sentMessages) this.trajectory.logSentMessage(message);
 
-    const response = await this.model.completionWithRetry({
-      messages: sentMessages,
-      model: this.model.modelName,
-      stream: true,
-      temperature: options?.temperature,
-    });
+    const response = await (isO1
+      ? // o1 currently doesn't support streaming or temperatures != 1
+        this.model.completionWithRetry({
```

Comment:
Do we know what langchains retry logic is if we hit our maximum RPM? Is there anything else we need to do here?

Reply:
Retry definitely important to understand.

Reply:
Langchain uses p-retry with 6 exponentially backing off retries by default; `onFailedAttempt()` above decides whether an error is fatal or retryable.  I'm not sure what OpenAI will return on rate limit, but this is where we'll need to adjust if we encounter issues.